### PR TITLE
[insteon] Use shared jetty http client

### PR DESCRIPTION
**Pull request by jsetton**
Use the `HttpClientFactory` shared Jetty HTTP client over `HttpURLConnection` to communicate with an Insteon Hub 2.

I did run a successful test in my environment of this change.

### CODE REVIEW

File: bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java
Comment (jlaur) [inline bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java:297]: For `InterruptedException`, restore the interrupted state of the thread by calling `Thread.currentThread().interrupt()`, see https://rules.sonarsource.com/java/RSPEC-2142/. You could also let the exception propagate up the call stack to make sure the binding won't react to the `IOException` with any retry mechanism or similar when requested to stop.

**Review by jlaur — COMMENTED** [review]
Thanks! Only one comment, otherwise LGTM.

**Comment by jsetton** [comment]
I added some refactoring of the `IOStream` classes since this PR is touching these classes. If you feel it's too much, I can submit a separate PR with these changes. I did run a successful test in my environment of this change.

**Comment by jlaur** [comment]
> I added some refactoring of the `IOStream` classes since this PR is touching these classes. If you feel it's too much, I can submit a separate PR with these changes.

I was getting ready to merge, so if you could extract that, I can merge today and you could create another PR. You can also leave it here, but I can't promise when I'll have time to review it, since I'm quite busy the next days. So you'll kind of be blocking yourself, since another maintainer with time at hand might be able to review the next smaller PR before I'll be able to. 😉

**Comment by jsetton** [comment]
> I was getting ready to merge, so if you could extract that, I can merge today and you could create another PR.

I removed the refactor commit and will include in separate PR. You should be able to merge. Thanks.

File: bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java
Comment (jlaur) [inline bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java:54]: Consider:
```suggestion
    private static final int REQUEST_TIMEOUT_SECONDS = 30;
```

This will make it harder to miss where used.

File: bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/LegacyPort.java
Comment (jlaur) [inline bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/LegacyPort.java:497]: This seems unrelated to the HTTP client refactoring, or maybe I'm missing something? What is the consequence of this not being here?

File: bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java
Comment (jsetton) [reply bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java:54]: There is a `TimeUnit` specified where it is used. The comment is more for context and consistent with how I defined similar constant in the binding.

https://github.com/openhab/openhab-addons/blob/e3ac1ad2b6709dbac6f1b738c6d81c2a52a46b08/bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java#L281-L282

File: bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/LegacyPort.java
Comment (jsetton) [reply bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/LegacyPort.java:497]: No consequence. The listener is only needed when the first message is sent to the modem. It didn't need to be added when the `LegacyPort` object is initialized.

File: bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java
Comment (jlaur) [reply bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java:54]: OK.

File: bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/LegacyPort.java
Comment (jlaur) [reply bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/LegacyPort.java:497]: I missed that it was previously in the constructor. Thanks!

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**
